src/main/java/org/apache/mesos/offer/RequirementUtils.java

```diff
@@ -36,13 +38,36 @@
   }
 
   public static Collection<ResourceRequirement> getResourceRequirements(Collection<Resource> resources) {
-    Collection<ResourceRequirement> resourceRequirements = new ArrayList<ResourceRequirement>();
+    Collection<ResourceRequirement> resourceRequirements = new ArrayList<>();
 
     for (Resource resource : resources) {
-      resourceRequirements.add(new ResourceRequirement(resource));
+      if (!isDynamicPort(resource)) {
+        resourceRequirements.add(new ResourceRequirement(resource));
+      }
     }
 
     return resourceRequirements;
   }
 
+  public static Collection<DynamicPortRequirement> getDynamicPortRequirements(List<Resource> resources)
+          throws DynamicPortRequirement.DynamicPortException {
+    Collection<DynamicPortRequirement> portRequirements = new ArrayList<>();
+
+    for (Resource resource : resources) {
+      if (isDynamicPort(resource)) {
+        portRequirements.add(new DynamicPortRequirement(resource));
+      }
+    }
+
+    return portRequirements;
+  }
+
+  static boolean isDynamicPort(Resource resource) {
+    if (resource.getName().equals("ports")) {
```

Comment:
(no work needed) I was going to recommend using a single SCALAR value instead of a RANGE, but then realized that SCALAR would be a double..